gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/activity/impl/DeleteWorkDirsActivityImpl.java

```diff
@@ -96,129 +68,26 @@ public CleanupResult cleanup(WUProcessingSpec workSpec, EventSubmitterContext ev
     }
   }
 
-  private boolean canCleanStagingData(JobContext jobContext, JobState jobState)
-      throws IOException {
-    return jobContext.getSemantics() != DeliverySemantics.EXACTLY_ONCE || !jobContext.getCommitSequenceStore()
-        .get().exists(jobState.getJobName());
+  private static Map<String, Long> cleanupStagingDataPerTask(JobState jobState, Set<String> resourcesToClean) throws IOException {
+    log.warn("Clean up staging data by task is not supported, will clean up job level data instead");
+    return cleanupStagingDataForEntireJob(jobState, resourcesToClean);
```

Comment:
I like this sentiment... just want to think about whether it should be stronger... such as throwing an exception rather than silently doing something different

Reply:
Right now the default in Gobblin is to clean staging dir at a task level, so I think rejecting the default can be a bit risky here?

Reply:
I see (NOW) that is the default... I didn't realize we've been operating w/ such widespread override of the default.

I do agree it's risky.  still that logging basically announces the default isn't supported and that an alternative approach is to be taken instead.  do you believe users would be more surprised by an exception or to discover that an unexpected form of cleanup was silently substituted?

I don't see a clear answer, so curious about your opinion.  to me, the silent switch-a-roo is more surprising.  my preference would be to fail-fast w/ an exception, but to define a property that could be used to override that behavior and do job-level deletion like you have here.  the difference is that the switch-up becomes opt-in (and explicit), not silently substituted.

Reply:
In this case wouldn't it be better not to have the additional configuration and have the user switch explicitly from clean up by task to clean up at a job level configuration?